File: src/model/course-admin/CourseTreeResponse.ts
import { CourseStatus } from "@/types/enum/CourseStatus";
import { CourseModuleResponse } from "./CourseModuleResponse";

export interface CourseTreeResponse {
  courseId: string;
  title: string;
  slug: string;
  description: string;
  isPublished: boolean;
  publishedAt?: string;
  status: CourseStatus;
  imageUrl?: string;
  videoUrl?: string;
  level?: string;
  category?: string;
  subCategory?: string;
  capacity?: number;
  requirements?: string;
  objectives?: string;
  syllabus?: string;
  shortDescription?: string;
  modules: CourseModuleResponse[];
  enrolledCount?: number;
  rating?: number;
  ratingCount?: number;
  updatedAt?: string;
}

File: src/main/java/com/truongsonkmhd/unetistudy/dto/course_dto/CourseTreeResponse.java
package com.truongsonkmhd.unetistudy.dto.course_dto;

import com.truongsonkmhd.unetistudy.common.CourseStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CourseTreeResponse {
    UUID courseId;
    String title;
    String slug;
    String description;
    String shortDescription;
    String level;
    String category;
    String subCategory;
    Integer duration;
    Integer capacity;
    String requirements;
    String objectives;
    String syllabus;
    List<String> learningOutcomes;
    java.time.LocalDateTime publishedAt;
    Boolean isPublished;
    CourseStatus status;
    String imageUrl;
    String videoUrl;
    String youtubeVideoId;
    String embedUrl;
    List<CourseModuleResponse> modules;

    Integer enrolledCount;
    Double rating;
    Integer ratingCount;
    Instant updatedAt;
}
